resources/providers/awslib/ecr/provider.go

```diff
@@ -15,56 +15,54 @@
 // specific language governing permissions and limitations
 // under the License.
 
-package awslib
+package ecr
 
 import (
 	"context"
 	"fmt"
 	"github.com/aws/aws-sdk-go-v2/aws"
-	"github.com/aws/aws-sdk-go-v2/service/ecr"
+	ecrClient "github.com/aws/aws-sdk-go-v2/service/ecr"
 	"github.com/aws/aws-sdk-go-v2/service/ecr/types"
+	"github.com/elastic/cloudbeat/resources/providers/awslib"
 )
 
-type EcrRepository types.Repository
-type EcrRepositories []types.Repository
-
-type EcrProvider struct {
-}
-
-type EcrRepositoryDescriber interface {
-	DescribeRepositories(ctx context.Context, cfg aws.Config, repoNames []string, region string) (EcrRepositories, error)
-}
-
-func NewEcrProvider() *EcrProvider {
-	return &EcrProvider{}
-}
-
 // DescribeAllEcrRepositories returns a list of all the existing repositories
-func (provider *EcrProvider) DescribeAllEcrRepositories(ctx context.Context, cfg aws.Config, region string) (EcrRepositories, error) {
+func (provider *Provider) DescribeAllEcrRepositories(ctx context.Context, cfg aws.Config, region string) ([]types.Repository, error) {
```

Comment:
```suggestion
func (provider *Provider) DescribeAllEcrRepositories(ctx context.Context, region string) ([]types.Repository, error) {
```

`cfg` is an unused parameter